### PR TITLE
[ch150283] date picker

### PR DESCRIPTION
**Pull request by mmmmmmaya**
### Summary
- **What:** 
  - Create a home base for filtering data (FilterPanel) -- all filtering logic stays in here
  - Create a basic UI component for filtering based on collection date
  - Separate out DateField ui/styles from upload-specific functionality for the DateField component. This allows us to reuse the basic functionality and styles for a DateField elsewhere in the app
- **Ticket:** [[ch150283]](https://app.clubhouse.io/genepi/story/150283)
- **Env:** https://featfiltering-frontend.dev.genepi.czi.technology/

### Testing
1. Test the indeterminate checkboxes still work 
1. Test the searching functionality still works
2. Ensure the filters return only the samples you would expect based on the given date range
3. Test a case where you only enter a start or end date

### Demos
https://user-images.githubusercontent.com/7562933/130283011-237e2aaf-244f-4444-bfdc-789bb749f109.mp4
https://user-images.githubusercontent.com/7562933/130283035-6733ce33-4066-4e6e-81b5-84f74c8ef874.mp4
https://user-images.githubusercontent.com/7562933/130283039-de66364d-25d2-4817-8532-2d7a991c8088.mp4

### TODO
- use new MenuSelect component from sds (Thanks, Timmy!) to get chips and styles
- styling to match design

### Checklist
- [x] I merged latest `feat/filtering`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/frontend/src/components/DateField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { FormikContextType } from "formik";
+import React from "react";
+import { StyledTextField } from "./style";
+
+// "YYYY-MM-DD".length
```
Comment (phoenixAja) [inline src/frontend/src/components/DateField/index.tsx]: can probs remove this lol

File: src/frontend/src/components/FilterPanel/components/CollectionDateFilter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import { Button, Menu, MenuItem } from "czifui";
+import { useFormik } from "formik";
+import { noop } from "lodash";
+import React, { FC, useEffect, useState } from "react";
+import DateField from "src/components/DateField";
+import {
+  DATE_ERROR_MESSAGE,
+  DATE_REGEX,
+} from "src/components/DateField/constants";
+import * as yup from "yup";
```
Comment (phoenixAja) [inline src/frontend/src/components/FilterPanel/components/CollectionDateFilter/index.tsx:10]: not sure how big this library is, but maybe would be good to only import what we need?

File: src/frontend/src/components/FilterPanel/components/CollectionDateFilter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import { StyledDateRange, StyledText } from "./style";
+
+type FromattedDateType = string | undefined;
```
Comment (phoenixAja) [inline src/frontend/src/components/FilterPanel/components/CollectionDateFilter/index.tsx]: should this be `FormattedDateType` ?

**Review by phoenixAja — APPROVED** [review]
this is working for me! Looks good! Not sure if we want to do a design review now or wait until later. Up to you!

File: src/frontend/src/components/FilterPanel/components/CollectionDateFilter/index.tsx (hunk 2 above)
Comment (mmmmmmaya) [reply src/frontend/src/components/FilterPanel/components/CollectionDateFilter/index.tsx:10]: A cool thing! You can check lib sizes on [bundlephobia](https://bundlephobia.com/package/yup@0.32.9). Also, afaict yup only has a default export, so importing only what we need isn't possible in the way you're suggesting (I think).

**Pull request merged**